crates/bevy_xr_utils/src/tracking_utils.rs

```diff
@@ -0,0 +1,314 @@
+use std::ops::Deref;
+
+use bevy::prelude::*;
+use bevy_mod_openxr::{
+    action_binding::{OxrSendActionBindings, OxrSuggestActionBinding},
+    action_set_attaching::OxrAttachActionSet,
+    action_set_syncing::{OxrActionSetSyncSet, OxrSyncActionSet},
+    helper_traits::{ToQuat, ToVec3},
+    resources::{OxrFrameState, OxrInstance, Pipelined},
+    session::OxrSession,
+    spaces::{update_space_transforms, OxrSpaceExt, OxrSpaceLocationFlags, OxrSpaceSyncSet},
+};
+use bevy_mod_xr::{
+    session::{session_available, session_running, XrSessionCreated, XrTrackingRoot},
+    spaces::{XrPrimaryReferenceSpace, XrReferenceSpace, XrSpace},
+    types::XrPose,
+};
+use openxr::Posef;
+
+//exernal api
+#[derive(Component)]
+pub struct XRTrackedStage;
+
+#[derive(Component)]
+pub struct XRTrackedLocalFloor;
+
+#[derive(Component)]
+pub struct XRTrackedView;
+
+#[derive(Component)]
+pub struct XRTrackedLeftGrip;
+
+#[derive(Component)]
+pub struct XRTrackedRightGrip;
```

Comment:
the naming convention is `Xr*` instead of `XR*` so here it would be `XrTracked*`

Reply:
I will rename these